packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/queue/queue.ts

```diff
@@ -0,0 +1,389 @@
+      return;
+    }
+    if (!this.parentService.scalableTaskCount) {
+      throw Error(`Auto scaling target for the service '${this.parentService.id}' hasn't been configured. Please use Service construct to configure 'minTaskCount' and 'maxTaskCount'.`);
+    }
+
+    this.addQueueScalingPolicy(this._eventsQueue, this._eventsQueueDelay);
+    for (const subsQueue of this.subscriptionQueues) {
+      if (subsQueue.scaleOnLatency) {
+        this.addQueueScalingPolicy(subsQueue.queue, subsQueue.scaleOnLatency);
+      } else {
+        this.addQueueScalingPolicy(subsQueue.queue, this._eventsQueueDelay);
+      }
+    }
+    this.parentService.enableAutoScalingPolicy();
+
+    this.createLambdaFunction(service);
+  }
+
+  /**
+   * This method adds a target tracking policy based on the backlog per task custom metric
+   * to the auto scaling target configured for this service.
+   *
+   * @param queue The queue for which backlog per task metric is being configured
+   * @param queueDelay The auto scaling options for the queue
+   */
+  private addQueueScalingPolicy(queue: sqs.IQueue, queueDelay: QueueAutoScalingOptions) {
+    const messageProcessingTime = queueDelay.messageProcessingTime.toSeconds();
+    const acceptableLatency = queueDelay.acceptableLatency.toSeconds();
+    if (messageProcessingTime > acceptableLatency) {
+      throw Error('Message processing time for the queue cannot be greater acceptable queue latency.');
+    }
+    const acceptableBacklog = acceptableLatency/messageProcessingTime;
+
+    this.parentService.scalableTaskCount?.scaleToTrackCustomMetric(`${queue.node.id}-autoscaling-policy`, {
+      metric: new cloudwatch.Metric({
+        namespace: 'Test',
+        metricName: 'BacklogPerTask',
+        dimensionsMap: { QueueName: queue.queueName },
+        unit: cloudwatch.Unit.COUNT,
+      }),
+      targetValue: acceptableBacklog,
+    });
+  }
+
+  /**
+   * This method is used to create the AWS Lambda Function for calculating backlog
+   * per task metric and a Cloudwatch event trigger for this function.
+   *
+   * @param service - The generated service.
+   */
+  private createLambdaFunction(service: ecs.Ec2Service | ecs.FargateService) {
+    const queueNames = [this._eventsQueue.queueName];
+    this.subscriptionQueues.forEach(subs => queueNames.push(subs.queue.queueName));
+
+    const backLogPerTaskCalculator = new lambda.Function(this.scope, 'BackLogPerTaskCalculatorFunction', {
+      runtime: lambda.Runtime.PYTHON_3_9,
+      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda')),
+      handler: 'index.queue_handler',
+      environment: {
+        CLUSTER_NAME: this.parentService.cluster.clusterName,
+        SERVICE_NAME: service.serviceName,
+        NAMESPACE: `${this.parentService.environment.id}-${this.parentService.id}`,
+        QUEUE_NAMES: queueNames.join(','),
+      },
+      initialPolicy: [new iam.PolicyStatement({
+        actions: ['ecs:DescribeServices'],
+        resources: ['*'],
```

Comment:
Can we not limit it to our current service because we don't know the ARN of the service yet?

Is there no way to create the Lambda function post-service creation?

Reply:
We're actually creating the function post-service creation so we can use the service ARN. Updated!!